src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.Cancun.cs

```diff
@@ -16,8 +14,66 @@ public partial class EngineRpcModule : IEngineRpcModule
 {
     private readonly IAsyncHandler<byte[], GetPayloadV3Result?> _getPayloadHandlerV3;
 
-    public Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV3(ExecutionPayload executionPayload) =>
-        NewPayload(executionPayload, 3);
+    public Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV3(ExecutionPayload executionPayload, byte[][] blobVersionedHashes)
+    {
+        if (blobVersionedHashes is null)
+        {
+            string error = "Blob versioned hashes must be set";
+            if (_logger.IsWarn) _logger.Warn(error);
+            return ResultWrapper<PayloadStatusV1>.Fail(error, ErrorCodes.InvalidParams);
+        }
+
+        if (executionPayload is null)
+        {
+            string error = "Execution payload must be set";
+            if (_logger.IsWarn) _logger.Warn(error);
+            return ResultWrapper<PayloadStatusV1>.Fail(error, ErrorCodes.InvalidParams);
+        }
+
+        int index = 0;
+        foreach (Transaction tx in executionPayload.GetTransactions())
+        {
+            if (tx.BlobVersionedHashes is null)
+            {
+                continue;
+            }
```

Comment:
Is this fine? Shouldn't `TxType` be checked? Can this be null for blob transaction? Can this be non-null for non-blob transaction?